### ISSUE TITLE
[java] LiteralsFirstInComparisons: False negative for methods returning Strings

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2569]
**Affects PMD Version:** 
6.24.0

**Description:**

Rule [LiteralsFirstInComparisons](
https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#literalsfirstincomparisons) does not flag the following :

```java
package com.javacodegeeks.patterns.chainofresponsibility;

public class ImageFileHandler implements Handler {
    @Override
  public void process(File file) {
    if (file.getFileType().equals("image")) { // violation expected
      System.out.println("Process and saving image file... by ");
      return;
    }
  }

}

```

**Running PMD through:** *[CLI | Ant ]*
